algorithms/kernel/pca/transform/oneapi/cl_kernels/pca_transform_cl_kernels.cl

```diff
@@ -30,12 +30,13 @@
 
 DECLARE_SOURCE(pca_transform_cl_kernels,
 
-__kernel void computeInvSigmas(__global algorithmFPType* rawVariances,
+__kernel void computeInvSigmas(__global const algorithmFPType* rawVariances,
                                __global algorithmFPType* invSigmas)
 {
     const int tid = get_global_id(0);
+    const int epsilon = 1e-10;
 
-    if (rawVariances[tid] != (algorithmFPType)0)
+    if (rawVariances[tid] != (algorithmFPType)epsilon)
```

Comment:
Please read about floating point comparison. In our case, `fabs(rawVariances[tid]) < epsilon` should be enough, I think